Sources/Segment/Settings.swift

```diff
@@ -79,10 +79,23 @@ extension Analytics {
     }
     
     internal func checkSettings() {
+        if isUnitTesting {
+            // we don't really wanna wait for this network call during tests...
+            // but we should make it work similarly.
+            store.dispatch(action: System.ToggleRunningAction(running: false))
+            DispatchQueue.main.async {
```

Comment:
Do we want `[weak self] in` for the closure?

Reply:
yes, good call out.